tsl/src/compression/compression.c

```diff
@@ -1640,12 +1640,16 @@ compress_singlerow(CompressSingleRowState *cr, TupleTableSlot *in_slot)
 			if (column->segment_info->is_null)
 				out_values[out_attrno] = 0;
 			else
-				out_values[out_attrno] = datumCopy(invalues[in_attrno],
-												   column->segment_info->typ_by_val,
-												   column->segment_info->typlen);
+				out_values[out_attrno] = invalues[in_attrno];
 		}
 		else
-			elog(ERRCODE_INTERNAL_ERROR, "unexpected state for column compressor");
+		{
+			/* we have a 1-1 column mapping from uncompressed -> compressed chunk.
+			 * However, some columns could have been dropped from the uncompressed
+			 *chunk before the compressed one is created.
+			 */
+			Assert(out_attrno == 0);
```

Comment:
To make it clear that 0 actually means that the attribute number is valid.
```suggestion
			Assert(out_attrno == InvalidAttrNumber);
```

Reply:
done

Reply:
This actually checks if we have a mapping col. These are initialized to 0. I'll rename this to out_colno so there is no confusion.